controllers/apply/lib/field-types.js

```diff
@@ -0,0 +1,153 @@
+'use strict';
+const get = require('lodash/fp/get');
+const { oneLine } = require('common-tags');
+const Joi = require('./joi-extensions');
+
+class TextField {
```

Comment:
This is the biggest change. Adds the concept of field types to give us a way of sharing common field logic. Aim is to make the standard form use these and then update the awards for all form to start using them once we've confirmed it works as an approach.